docs: sentry-go serverless usage and integrations removal (#1108)
getsentry/sentry-docs: src/collections/_documentation/platforms/go/serverless.md

---
title: Serverless
sidebar_order: 5
---

### Source Context

`sentry-go` SDK comes with the support for serverless solutions out of the box.

> However, in order to get the source contexts right, you need to bundle your source code with the binary itself.

Reviewer: ```suggestion
However, to get the source contexts right, you need to bundle your source code with the binary itself.
```

For example, when using __AWS Lambda__ and given this tree structure:

```bash
.
├── bin
│   └── upload-image
│   └── process-image
│   └── create-thumbnails
├── functions
│   └── upload-image
│       └── main.go
│   └── process-image
│       └── main.go
│   └── create-thumbnails
│       └── main.go
├── helper
│   ├── foo.go
│   └── bar.go
├── util
│   ├── baz.go
│   └── qux.go

```

> You can build one of the binaries and bundle them with necessary source files with a command:

Reviewer: ```suggestion
You can build one of the binaries and bundle them with the necessary source files with this command:
```

```bash
$ GOOS=linux go build -o bin/upload-image functions/upload-image/main.go && zip -r handler.zip bin/upload-image functions/upload-image/ helper/ util/
```

> The only requirement is that the source code is actually located on the deployed machine. Everything else should be done automatically by the SDK.

Reviewer: ```suggestion
The only requirement is that you locate the source code on the deployed machine. Everything else should be done automatically by the SDK.
```

### Events Delivery

Most (if not all) serverless solutions won't wait for network responses before closing the process.

> Because of that, we need to make sure ourselves that the event is delivered to Sentrys severs.

Reviewer: ```suggestion
Because of that, we need to make sure that the event is delivered to Sentry's severs.
```

> It can be achieved twofold, using `sentry.Flush` method or by swapping the transport to `HTTPSyncTransport`.

Reviewer: ```suggestion
It can be achieved twofold, using the `sentry.Flush` method or by swapping the transport to `HTTPSyncTransport`.
```
